custom_components/meross_cloud/sensor.py

```diff
@@ -247,6 +248,45 @@ def native_value(self) -> StateType:
     def should_poll(self) -> bool:
         return True
 
+class EnergySensorWrapper(GenericSensorWrapper):
+    _device: ElectricitySensorDevice
+
+    def __init__(self, device: ElectricitySensorDevice,
+                 device_list_coordinator: DataUpdateCoordinator[Dict[str, HttpDeviceInfo]], channel: int = 0):
+        super().__init__(sensor_class=DEVICE_CLASS_ENERGY,
+                         measurement_unit="kWh",
+                         device_method_or_property='async_get_daily_power_consumption',
+                         state_class=STATE_CLASS_TOTAL_INCREASING,
+                         device=device,
+                         device_list_coordinator=device_list_coordinator,
+                         channel=channel)
+
+        # Device properties
+        self._daily_consumption = None
+
+    # For ElectricityMixin devices we need to explicitly call the async_Get_instant_metrics
+    async def async_update(self):
+        if self.online:
+            await super().async_update()
+
+            if isinstance(self._device, ConsumptionXMixin):
```

Comment:
This won't be necessary any longer if we only create the wrapper around ConsumptionXMixin devices.